### 상관관계 확인

In [2]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity="all"

In [3]:
import numpy as np
import pandas as pd

df = pd.read_csv('./kaggle/input/spaceship-titanic/train_update.csv')
df01 = pd.read_csv('./kaggle/input/spaceship-titanic/train.csv')

In [ ]:
df_died = df[df['Transported']==0]

In [44]:
# # Cabin_port
# df_died.groupby('Cabin_port').count()
# df.groupby('Cabin_port').count()
# # HomePlanet
# df_died.groupby('HomePlanet').count()
# df.groupby('HomePlanet').count()
# # CryoSleep
# df_died.groupby('CryoSleep').count()
# df.groupby('CryoSleep').count()
# # Cabin
df_died.groupby('Cabin').count()
df.groupby('Cabin').count()

len(df_died.groupby('Cabin'))

PassengerId  HomePlanet  CryoSleep  Cabin_num  Cabin_port  Destination  \
Cabin                                                                           
0              129         129        129        129         129          129   
1              207         207        207        207         207          207   
2              239         239        239        239         239          239   
3              282         282        282        282         282          282   
4              630         630        630        630         630          630   
5             1583        1583       1583       1583        1583         1583   
6             1240        1240       1240       1240        1240         1240   
7                5           5          5          5           5            5   

        Age   VIP  RoomService  FoodCourt  ShoppingMall   Spa  VRDeck  Name  \
Cabin                                                                         
0       129   129          129        129           129   129     129   125   
1       207   207          207        207           207   207     207   203   
2       239   239          239        239           239   239     239   234   
3       282   282          282        282           282   282     282   278   
4       630   630          630        630           630   630     630   614   
5      1583  1583         1583       1583          1583  1583    1583  1541   
6      1240  1240         1240       1240          1240  1240    1240  1217   
7         5     5            5          5             5     5       5     4   

       Transported  
Cabin               
0              129  
1              207  
2              239  
3              282  
4              630  
5             1583  
6             1240  
7                5

PassengerId  HomePlanet  CryoSleep  Cabin_num  Cabin_port  Destination  \
Cabin                                                                           
0              256         256        256        256         256          256   
1              779         779        779        779         779          779   
2              747         747        747        747         747          747   
3              495         495        495        495         495          495   
4             1025        1025       1025       1025        1025         1025   
5             2821        2821       2821       2821        2821         2821   
6             2564        2564       2564       2564        2564         2564   
7                6           6          6          6           6            6   

        Age   VIP  RoomService  FoodCourt  ShoppingMall   Spa  VRDeck  Name  \
Cabin                                                                         
0       256   256          256        256           256   256     256   249   
1       779   779          779        779           779   779     779   763   
2       747   747          747        747           747   747     747   733   
3       495   495          495        495           495   495     495   485   
4      1025  1025         1025       1025          1025  1025    1025  1005   
5      2821  2821         2821       2821          2821  2821    2821  2747   
6      2564  2564         2564       2564          2564  2564    2564  2506   
7         6     6            6          6             6     6       6     5   

       Transported  
Cabin               
0              256  
1              779  
2              747  
3              495  
4             1025  
5             2821  
6             2564  
7                6

8

In [54]:
df_died.groupby('Cabin_num').count()['PassengerId'][354]

3

In [56]:
# 비율 확인

def died_persent(column_name):
    
    for i in range(len(df_died.groupby(column_name))):
        try:
            df_d0 = df_died.groupby(column_name).count()['PassengerId'][i] 
            df_0 = df.groupby(column_name).count()['PassengerId'][i]  
            print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
        except:
            pass

died_persent('Cabin_num')

Cabin_num 0 : 전체 22 중 14명 63.64% 실종
Cabin_num 1 : 전체 15 중 5명 33.33% 실종
Cabin_num 2 : 전체 11 중 6명 54.55% 실종
Cabin_num 3 : 전체 16 중 5명 31.25% 실종
Cabin_num 4 : 전체 7 중 5명 71.43% 실종
Cabin_num 5 : 전체 13 중 6명 46.15% 실종
Cabin_num 6 : 전체 12 중 8명 66.67% 실종
Cabin_num 7 : 전체 9 중 3명 33.33% 실종
Cabin_num 8 : 전체 13 중 7명 53.85% 실종
Cabin_num 9 : 전체 16 중 4명 25.0% 실종
Cabin_num 10 : 전체 14 중 8명 57.14% 실종
Cabin_num 11 : 전체 19 중 8명 42.11% 실종
Cabin_num 12 : 전체 5 중 2명 40.0% 실종
Cabin_num 13 : 전체 16 중 13명 81.25% 실종
Cabin_num 14 : 전체 17 중 5명 29.41% 실종
Cabin_num 15 : 전체 10 중 3명 30.0% 실종
Cabin_num 16 : 전체 17 중 3명 17.65% 실종
Cabin_num 17 : 전체 8 중 6명 75.0% 실종
Cabin_num 18 : 전체 18 중 6명 33.33% 실종
Cabin_num 19 : 전체 22 중 9명 40.91% 실종
Cabin_num 20 : 전체 15 중 9명 60.0% 실종
Cabin_num 21 : 전체 27 중 17명 62.96% 실종
Cabin_num 22 : 전체 15 중 11명 73.33% 실종
Cabin_num 23 : 전체 16 중 10명 62.5% 실종
Cabin_num 24 : 전체 16 중 7명 43.75% 실종
Cabin_num 25 : 전체 15 중 6명 40.0% 실종
Cabin_num 26 : 전체 10 중 1명 10.0% 실종
Cabin_num 27 : 전체 17 중 9명 52.94% 실종
Cabin_num

In [38]:
# # 비율 확인

# def died_persent(column_name):
#     df_d0 =df_died.groupby(column_name).count()['PassengerId'][0] #
#     df_d1 = df_died.groupby(column_name).count()['PassengerId'][1] #
#     df_0 = df.groupby(column_name).count()['PassengerId'][0] # 
#     df_1 = df.groupby(column_name).count()['PassengerId'][1] #
#     print(f'{column_name} 0 : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
#     print(f'{column_name} 1 : 전체 {df_1} 중 {df_d1}명 {round(df_d1/df_1*100,2)}% 실종')

# died_persent('CryoSleep')

CryoSleep 0 : 전체 5568 중 3741명 67.19% 실종
CryoSleep 1 : 전체 3125 중 574명 18.37% 실종


In [29]:
574/3125*100

18.368000000000002

### 도착하지 못한 사람
Transported == 0
1. Cabin_port
    - 좌현(P,0) 전체 4403 중 2405 명 54.6% 실종 
    - 우현(S,0) 전체 4290 중 1910 명 44.5% 실종
2. HomePlanet
    - 비슷한 비율로 실종된것 처럼 보인다 
    - Earth:0 : 전체 4714 중 2719 명 57.7% 실종 
    - Mars:1 : 전체 2175 중 740 명 34.0% 실종 
    - Europa:2 : 전체 1804 중 856 명 47.5% 실종 
3. CryoSleep(방 밖으로 나가지 않는 사람)
    - CryoSleep 0 : 전체 5568 중 3741명 67.19% 실종
    - CryoSleep 1 : 전체 3125 중 574명 18.37% 실종
4. Cabin
    - Cabin 0 : 전체 256 중 129명 50.39% 실종
    - Cabin 1 : 전체 779 중 207명 26.57% 실종
    - Cabin 2 : 전체 747 중 239명 31.99% 실종
    - Cabin 3 : 전체 495 중 282명 56.97% 실종
    - Cabin 4 : 전체 1025 중 630명 61.46% 실종
    - Cabin 5 : 전체 2821 중 1583명 56.11% 실종
    - Cabin 6 : 전체 2564 중 1240명 48.36% 실종
    - Cabin 7 : 전체 6 중 5명 83.33% 실종
    

모든 퍼센트는 소수 2째자리 반올림

### 저체온증 관련

선장이 식당을 제외한 모든 곳에서 난방과 온수 시스템을 끄라고 지시 -> FoodCourt 와 실종과의 관련성
[해당 링크](https://namu.wiki/w/%ED%83%80%EC%9D%B4%ED%83%80%EB%8B%89%ED%98%B8%20%EC%B9%A8%EB%AA%B0%20%EC%82%AC%EA%B3%A0)